integration/network/link_test.go

```diff
@@ -0,0 +1,100 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDockerNetworkLinkAlias(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	mynet, err := client.NetworkCreate(ctx, "mynet", types.NetworkCreate{})
+	require.NoError(t, err)
+
+	nws, err := client.NetworkList(ctx, types.NetworkListOptions{})
+	require.NoError(t, err)
+	assert.Equal(t, true, containsNetwork(nws, mynet), "failed to create network mynet")
+
+	foo, err := client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"top"},
+			Image: "busybox",
+		},
+		&container.HostConfig{
+			NetworkMode: "mynet",
+		},
+		&network.NetworkingConfig{
+			EndpointsConfig: map[string]*network.EndpointSettings{
+				"mynet": {},
+			},
+		},
+		"foo",
+	)
+	require.NoError(t, err)
+
+	err = client.ContainerStart(ctx, foo.ID, types.ContainerStartOptions{})
+	require.NoError(t, err)
+
+	bar, err := client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"ping", "-c", "3", "server"},
+			Image: "busybox",
+		},
+		&container.HostConfig{
+			NetworkMode: "mynet",
+			Links:       []string{fmt.Sprintf("%s:server", foo.ID)},
+		},
+		&network.NetworkingConfig{
+			EndpointsConfig: map[string]*network.EndpointSettings{
+				"mynet": {
+					Links: []string{fmt.Sprintf("%s:server", foo.ID)},
+				},
+			},
+		},
+		"bar",
+	)
+	require.NoError(t, err)
+
+	err = client.ContainerStart(ctx, bar.ID, types.ContainerStartOptions{})
+	require.NoError(t, err)
+	poll.WaitOn(t, containerIsStopped(ctx, client, bar.ID), poll.WithDelay(100*time.Millisecond))
+
+	body, err := client.ContainerLogs(ctx, "bar", types.ContainerLogsOptions{
+		ShowStdout: true,
+		//ShowStderr: true,
```

Comment:
can you remove this one?

Reply:
Thanks @thaJeztah. The PR has been updated.